src/schema/formats.js

```diff
@@ -103,6 +103,41 @@ export function isSecureUrl(value) {
   return ['https:', 'wss:'].includes(url.protocol);
 }
 
+export function isOrigin(value) {
+  // FIXME: should we only have isSecureOrigin ?
+  let url;
+  try {
+    url = new URL(value);
+  } catch (e) {
+    // It's invalid or not absolute.
+    return false;
+  }
+  if (value.includes('*')) {
+    // Wildcards are not valid in origins.
+    return false;
+  }
```

Comment:
We may not need to explicitly check for "*" given that this is an origin and we are already going to require that the string is strictly equal to the origin (and with that check we should already excluding a number of url components where `new URL(...)` would not be complaining about a "*"), that should be leaving only protocol, hostname and protocol and new URL should be throwing already with a "*" placed there.

@diox do you see any issue in trying to match even more the kind of checks that the patch on the Firefox side is going to be doing? is there any case covered by this method but not by the one in Firefox?

In the current version of that patch, the origin formatted does look like in the following snippet (pasting here because phabricator does not allow easily to link a certain like in a revision):
```
  origin(string, context) {
    let url;
    try {
      url = new URL(string);
    } catch (e) {
      throw new Error(`Invalid origin: ${string}`);
    }
    if (!/^https?:/.test(url.protocol)) {
      throw new Error(`Invalid origin must be http or https for URL ${string}`);
    }
    // url.origin is punycode so a direct check against string wont work.
    // url.href appends a slash even if not in the original string, we we
    // additionally check that string does not end in slash.
    if (string.endsWith("/") || url.href != new URL(url.origin).href) {
      throw new Error(
        `Invalid origin for URL ${string}, replace with origin ${url.origin}`
      );
    }
    if (!context.checkLoadURL(url.origin)) {
      throw new Error(`Access denied for URL ${url}`);
    }
    return url.origin;
  },
``` 

Besides the returned value (and the call to context.checkLoadURL, which it shouldn't matter nor be needed for the addons-linter validation check), it looks we may be able to derive the addons-linter validation function with a almost 1-1 match with the Firefox checks.

wdyt?

Reply:
Yes, I haven't had the time to do that yet but I plan to.

Reply:
Did that in https://github.com/mozilla/addons-linter/pull/4018/commits/86a1a32e5a314e9a7f3c64b3a02a2bd529c36e7d (and https://github.com/mozilla/addons-linter/pull/4018/commits/104f477a3e7d81d19e595aaa9534ac8ed9d3746c to make `eslint` happy).

I kept the logic but had to make a few changes:
- Changed the `throw`s for `return false`
- Returned `true` instead of `url.origin` at the end
- Dropped the `context` variable and the `context.checkLoadURL` call
- Add something for the wildcards (which is more for convenience so that developers don't try to use them only to realize it doesn't work that way when they try to install their add-on later). 
- Renamed `string` to `value` for consistency with the rest of the file